native-pkcs11-core/src/object.rs

```diff
@@ -74,6 +75,33 @@ impl Clone for Object {
     }
 }
 
+fn extract_ec_params(der_bytes: &[u8]) -> Option<(Vec<u8>, Vec<u8>)> {
+    let spki: SubjectPublicKeyInfoRef<'_> = SubjectPublicKeyInfoRef::try_from(der_bytes).unwrap();
+    // For EC keys, the algorithm parameters contain the curve OID
+    // For EC keys, the subject public key is the EC point
+    Some((
+        ObjectIdentifier::from_bytes(spki.algorithm.parameters.unwrap().value())
+            .unwrap()
+            .to_der()
+            .unwrap(),
+        OctetString::new(spki.subject_public_key.raw_bytes()).unwrap().to_der().unwrap(),
+    ))
+}
+
+fn extract_rsa_params(der_bytes: &[u8]) -> Option<(Vec<u8>, Vec<u8>, u64)> {
+    // Parse the DER-encoded SPKI
+    let spki: SubjectPublicKeyInfoRef<'_> = SubjectPublicKeyInfoRef::try_from(der_bytes).unwrap();
+    // Parse the RSA public key bytes from the SPKI
+    let rsa_pubkey = RsaPublicKey::from_der(spki.subject_public_key.raw_bytes()).ok()?;
+    let modulus = rsa_pubkey.modulus.as_bytes();
```

Comment:
Nit: call `rsa_pubkey.modulus.as_bytes().to_vec()` in the struct constructor instead of assigning to a variable.